helpers/deployer/stages/validate.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package stages
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/mitchellh/go-testing-interface"
+
+	"github.com/terraform-google-modules/terraform-example-foundation/helpers/deployer/gcp"
+)
+
+const (
+	replaceME     = "REPLACE_ME"
+	exampleDotCom = "example.com"
+)
+
+// ValidateDirectories checks if the required directories exist
+func ValidateDirectories(g GlobalTfvars) error {
+	_, err := os.Stat(g.FoundationCodePath)
+	if os.IsNotExist(err) {
+		return fmt.Errorf("# Stopping execution, FoundationCodePath directory '%s' does not exits\n", g.FoundationCodePath)
+	}
+	_, err = os.Stat(g.CodeCheckoutPath)
+	if os.IsNotExist(err) {
+		return fmt.Errorf("# Stopping execution, CodeCheckoutPath directory '%s' does not exits\n", g.CodeCheckoutPath)
+	}
+	return nil
+}
+
+// ValidateBasicFields validates if the values for the required field were provided
+func ValidateBasicFields(t testing.TB, g GlobalTfvars) {
+	gcpConf := gcp.NewGCP()
+	fmt.Println("")
+	fmt.Println("# Validating tfvar file.")
+	if g.OrgID == replaceME {
+		fmt.Println("# Replace value for input 'org_id'")
+	} else {
+		if g.HasValidatorProj() && gcpConf.HasSccNotification(t, g.OrgID, g.SccNotificationName) {
+			fmt.Printf("# Notification '%s' exists in organization '%s'. Chose a different one.\n", g.SccNotificationName, g.OrgID)
+			fmt.Printf("# See existing Notifications for organization '%s'.\n", g.OrgID)
+			fmt.Printf("# gcloud scc notifications list organizations/%s --filter=\"name:organizations/%s/notificationConfigs/%s\" --format=\"value(name)\"\n", g.OrgID, g.OrgID, g.SccNotificationName)
+			fmt.Println("")
+		}
+		if g.HasValidatorProj() && !g.CreateUniqueTagKey && gcpConf.HasTagKey(t, g.OrgID, "environment") {
+			fmt.Printf("# Tag key 'environment' exists in organization '%s'.\n", g.OrgID)
+			fmt.Println("# Set variable 'create_unique_tag_key' to 'true' in the tfvar file.")
+			fmt.Println("")
+		}
+	}
+
+	if g.BillingAccount == replaceME {
+		fmt.Println("# Replace value for input 'billing_account'")
+	}
```

Comment:
nit: we could create a function for this that just looks up this struct tag using https://pkg.go.dev/reflect#StructField so this could become something like hasString() or containsString()

Reply:
Great idea.